company-tng.el

```diff
@@ -106,12 +106,15 @@ confirm the selection and finish the completion."
      (let ((ov (make-overlay (point) (point))))
        (setq company-tng--overlay ov)
        (overlay-put ov 'priority 2))
-     (advice-add 'company-select-next :before-until 'company-tng--allow-unselected)
-     (advice-add 'company-fill-propertize :filter-args 'company-tng--adjust-tooltip-highlight))
+     (setq company-selection-default nil)
+     (unless company-selection-changed
+       (setq company-selection company-selection-default))
+     (advice-add 'company-select-next :before-until 'company-tng--allow-unselected))
     (update
-     (let ((ov company-tng--overlay)
-           (selected (nth company-selection company-candidates))
-           (prefix (length company-prefix)))
+     (let* ((ov company-tng--overlay)
+            (selection (or company-selection 0))
```

Comment:
I'm guessing the `or` here is actually unnecessary. Or should be, because this action should only be possible once the user has selected something.

We could add an assert here, though. For confidence.

Reply:
Done, I've checked the `selected` with `company-selection`